portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -43,8 +43,19 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     PreparedQuery results = fetch(); 
     List<Comment> comments = toInternal(results);
     String filter = getSortingCriterion(request);
-    sort(comments, filter);
-    String json = toJson(comments);
+    int quantity = getNumberComment(request);
+    sort(comments, filter); 
+
+    if(quantity > comments.size()) {
+        quantity = comments.size();
+    }
+    
+    List<Comment> limitedComments = new ArrayList<Comment>();
```

Comment:
Can't you use https://docs.oracle.com/javase/8/docs/api/java/util/List.html#subList-int-int-?